src/builder.rs

```diff
@@ -231,6 +233,75 @@ impl<'w, 's, 'a> Builder<'w, 's, 'a> {
         )
     }
 
+    /// Create an operation that unzips its inputs and sends each element off to
+    /// a different output.
+    pub fn create_unzip<T>(&mut self) -> (InputSlot<T>, T::Unzipped)
+    where
+        T: Unzippable + 'static + Send + Sync,
+    {
+        let source = self.commands.spawn(()).id();
+        (
+            InputSlot::new(self.scope, source),
+            T::unzip_output(Output::<T>::new(self.scope, source), self),
+        )
+    }
+
+    /// Create an operation that creates a fork for a [`Result`] input. The value
+    /// inside the [`Result`] will be unpacked and sent down a different branch
+    /// depending on whether it was in the [`Ok`] or [`Err`] variant.
+    pub fn create_fork_result<T, E>(&mut self) -> (InputSlot<Result<T, E>>, ForkResultOutput<T, E>)
+    where
+        T: 'static + Send + Sync,
+        E: 'static + Send + Sync,
+    {
+        let source = self.commands.spawn(()).id();
```

Comment:
What is the implication of this (if any) in run time? Does creating another entity affect run time performance in any way compared to chaining?

Reply:
Eventually this `InputSlot` will get connected to some `Output`, and at that point the [`try_connect`](https://github.com/open-rmf/bevy_impulse/blob/2d9fe36f0042d2e2ba42e2a5087b19b4c057d3ed/src/builder/connect.rs#L58) command will run, which represents all the overhead for doing it this way. Since it only runs once while the workflow is being built, the runtime overhead is extremely negligible.

Reply:
Is there any difference when the workflow is running? Does the additional connection cause an additional "hop", resulting in additional copying/moving of messages?

Reply:
Nope, the `Connect` command works by replacing all references to the `Output` entity with the `InputSlot` entity, so there will be no difference in the final execution behavior.